pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/DeleteAPIHybridClusterIntegrationTest.java

```diff
@@ -0,0 +1,324 @@
+ * and at least 3 offline segments.
+ *
+ */
+public class DeleteAPIHybridClusterIntegrationTest extends HybridClusterIntegrationTest {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RealtimeClusterIntegrationTest.class);
+  private String TABLE_NAME;
+  private long nOfflineRows;
+  private long nRealtimeRows;
+  ControllerConf config;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    super.setUp();
+    TABLE_NAME = super.getTableName();
+    nOfflineRows = numRowsReturned(CommonConstants.Helix.TableType.OFFLINE);
+    nRealtimeRows = numRowsReturned(CommonConstants.Helix.TableType.REALTIME);
+    config = ControllerTestUtils.getDefaultControllerConfiguration();
+  }
+
+  private long numRowsReturned(CommonConstants.Helix.TableType tableType) throws Exception {
+    org.json.JSONObject response = postQuery("select count(*) from '" + TABLE_NAME + "_" + tableType + "'");
+    if (response.get("numDocsScanned").equals(new Integer(0))) {
+      return 0;
+    }
+    else {
+      // Throws a null pointer exception when there are no rows because it can't find "aggregationResults"
+      String pinotValue = ((org.json.JSONArray) response.get("aggregationResults")).getJSONObject(0).get("value").toString();
+      return Long.parseLong(pinotValue);
+    }
+  }
+
+  private boolean waitForNumRows(long numRows, CommonConstants.Helix.TableType tableType) throws Exception {
+    long start = System.currentTimeMillis();
+    long end = start + 60 * 1000;
+    while (System.currentTimeMillis() < end) {
+      if (numRowsReturned(tableType) == numRows) {
+        return true;
+      }
+      Thread.sleep(600);
+    }
+    Assert.fail("Operation took too long");
+    return false;
+  }
+
+  private long getNumRows(CommonConstants.Helix.TableType type) throws Exception{
+    org.json.JSONObject response = postQuery("select count(*) from '" + TABLE_NAME + "_" + type + "'");
+
+    long start = System.currentTimeMillis();
+    long end = start + 60 * 1000;
+    while (System.currentTimeMillis() < end) {
+      if (!response.get("numDocsScanned").equals(new Integer(0))) {
+        // Throws a null pointer exception when there are no rows because it can't find "aggregationResults"
+        String pinotValue = ((org.json.JSONArray) response.get("aggregationResults")).getJSONObject(0).get("value").toString();
+        return Long.parseLong(pinotValue);
+      }
+      Thread.sleep(200);
+    }
+    return 0L;
+  }
+  @Override // Leaving this out because it is done in the superclass
+  public void testGeneratedQueriesWithMultiValues() {}
+
+  @Override // Leaving this out because it is done in the superclass
+  public void testHardcodedQuerySet() {}
+
+  @Override // Leaving this out because it is done in the superclass
+  public void testBrokerDebugOutput() {}
+
+  @Test
+  public void deleteRealtimeSegmentFromGetAPI() throws Exception {
+    long currRealtimeRows = getNumRows(CommonConstants.Helix.TableType.REALTIME);
```

Comment:
Assuming tests start with all segments loaded, we don't need to wait when we get the current realtime rows.